### PR TITLE
Feature/create post page version2

### PR DESCRIPTION
**Pull request by nikidkny**
- made the create ensemblePosts work
- can see all the posts with related ensembles
- the posts can be filtered by instruments
- on profile page the user can see their own posts

right now I am fetching the /posts endpoint where I get all the posts that I combine with the ensemble data this result in an object {post,ensemble}
then in the frontend rather than having to make separate requests for it I get it with one request

### CODE REVIEW

File: client/src/pages/EnsemblesPage.tsx [hunk 1]
```diff
@@ -1,60 +1,101 @@
...
 import { InstrumentType } from "../types/InstrumentType";
 import Button from "../components/atoms/Button";
 import { ICON_NAMES } from "../components/atoms/Icon/IconNames";
-import EnsembleCard from "../components/molecules/EnsembleCard";
+import PostCard from "../components/molecules/PostCard";
+// import { EnsemblePostsType } from "../types/EnsemblePostsType";
+// import { EnsembleType } from "../types/EnsembleType";
+import { PostWithEnsembleType } from "../types/PostWithEnsembleType";
 
 //TODO: this page will be ensembles posts page and will need to be reworked.
```
Comment (andreaDiClaudio) [inline client/src/pages/EnsemblesPage.tsx:15]: Is this TODO done? do we need to change the name of the file ???

Comment (nikidkny) [reply client/src/pages/EnsemblesPage.tsx:15]: I wasn't sure, but in my opinion as there is only one type of post and we don't just have a collection of ensembles its okay to name it like how it is now

**Review by andreaDiClaudio — APPROVED** [review]
Can you please leave TODO for validation ( if you already did sorry didn't see them or thought they were missign in some places :) )

File: client/src/pages/EnsemblesPage.tsx (hunk 1 above)
Comment (amorea9) [reply client/src/pages/EnsemblesPage.tsx:15]: It was a WIP when I handed it over. This page has now the posts.. We could rename it to avoid confusion

**Review by amorea9 — DISMISSED** [review]

File: client/src/pages/CreateEnsemblePostPage.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+          <Button
+            buttonVariant="primary"
+            buttonLabel="Create post"
+            buttonState={(loading && "disabled") || "default"}
+            iconPosition="top"
+            className="w-auto m-b-6 py-4"
+            type="submit"
+          >
+            {loading ? "Creating..." : "Create post"}
```
[the remark this answers is not in the document]
Comment (nikidkny) [reply client/src/pages/CreateEnsemblePostPage.tsx:313]: We don't have to as that loading if the page is slow is still working

**Review by amorea9 — DISMISSED** [review]

File: server/src/modules/posts/posts.service.ts [hunk 3]
```diff
@@ -2,17 +2,52 @@ import { Injectable } from '@nestjs/common';
...
+      .exec();
+
+    // Map posts with their ensembles
+    return posts.map((post) => {
+      const ensemblePost = ensemblePosts.find(
+        (ep) => ep.postId.toString() === post._id.toString(),
+      );
+      return {
+        post,
+        ensemble: ensemblePost?.ensembleId || null,
```
[the remark this answers is not in the document]
Comment (nikidkny) [reply server/src/modules/posts/posts.service.ts:36]: You are right I removed that

File: client/src/pages/EnsemblesPage.tsx [hunk 4]
```diff
@@ -1,60 +1,101 @@
...
+    // ensemblePostsData,
+    // fetchEnsemblePosts,
+    // ensemblesData,
+  ]);
 
-  console.log(instrumentsFetch.data);
-  console.log(filterOption);
+  // Ensure filteredPosts is set correctly on initial load
+  useEffect(() => {
+    if (PostWithEnsembleData.length === 0) {
```
[the remark this answers is not in the document]
Comment (nikidkny) [reply client/src/pages/EnsemblesPage.tsx]: I tried but for some reason if I do not have it separate on first page load it does not load the postsWithEnsembleData

**Review by amorea9 — APPROVED** [review]

**Pull request merged**